pkg/provider/consulcatalog/consul_catalog.go

```diff
@@ -279,7 +334,19 @@ func (p *Provider) fetchServices(ctx context.Context) ([]string, error) {
 			continue
 		}
 
-		filtered = append(filtered, svcName)
+		connect := p.ConnectByDefault
+		if contains(tags, p.Prefix+".connect=true") {
+			connect = true
+		} else if contains(tags, p.Prefix+".connect=false") {
+			connect = false
+		}
+
```

Comment:
```suggestion
		connect := contains(tags, p.Prefix+".connect=true")
```

Reply:
I don't think this is equivalent. If you did that, you would completely ignore the value of p.ConnectByDefault. Is it what we want to do?

Reply:
No this is not what we want to do and still something from earlier revisions (same as above)